Assets/Shopify/UIToolkit/Shops/Interfaces/IShop.cs

```diff
@@ -53,10 +53,22 @@ public interface IShop {
         void OnPurchaseFailed(ShopifyError error);
 
         /// <summary>
-        /// Called when the total quantity of items in the cart changes.
-        /// Useful for when showing a badge next to the cart button for example.
+        /// Called when there was a change in the total number of items in the cart.
         /// </summary>
-        /// <param name="newQuantity">The new number of items in the cart</param>
-        void OnCartQuantityChanged(int newQuantity);
+        /// <param name="totalNumberOfCartItems"></param>
+        void OnCartQuantityChanged(int totalNumberOfCartItems);
+
+        /// <summary>
+        /// Called when the Items in the cart have updated. 
+        /// </summary>
+        /// <param name="lineItems">The new line items in the cart</param>
+        void OnCartItemsChanged(List<CartItem> lineItems);
+
+        /// <summary>
+        /// Call to update the quantity of the given variant in the cart.
+        /// </summary>
+        /// <param name="variant">Variant object that is currently in the user's Cart.</param>
+        /// <param name="quantity">The quantity value to change the cart item to.</param>
+        void UpdateCartQuantityForVariant(ProductVariant variant, long quantity);
```

Comment:
There was some discussion in a previous PR about keeping the Cart callbacks on the `IMultiProductShop` interface instead of `IShop` but there are a few issues I ran into in doing so:

1. The `CartController` lives in the `ShopControllerBase` which means the cart is accessible to all shops. If we wanted to change this we'd probably want to move it to a `MultiShopControllerBase` or something.

2. Even though a single product shop might not technically need a cart _view_, the concept of adding to a cart behind the scenes is still valid. There's also the case of adding/buying multiple items from a single product to consider.